lib/plugins/aws/deploy/lib/checkForChanges.js

```diff
@@ -98,7 +98,11 @@ module.exports = {
     });
 
     return BbPromise.all(getFunctionResults)
-      .then(_.min)
+      .then(results => {
+        if (!results || results.length === 0) return undefined;
+        const timestamp = Math.min(...results); // Math.min converts Date to timestamp
+        return new Date(timestamp);
```

Comment:
Let's maybe rely on `reduce` instead (?)

```javascript
results.reduce((currentMin, date) => (!currentMin || date < currentMin) ? date : currentMin, undefined)
```

Reply:
I'll try to make this part cleaner.